src/main/java/io/github/mzmine/modules/io/import_rawdata_mzxml/MzXMLImportTask.java

```diff
@@ -431,42 +398,30 @@ public void endElement(String namespaceURI, String sName, // simple name
             }
 
             // Copy m/z and intensity data
-            dps[i] = new SimpleDataPoint(mz, intensity);
-          }
-          // sort because old converters might create unsorted spectral data
-          Arrays.sort(dps, DataPointSorter.DEFAULT_MZ_ASCENDING);
-
-          for (int i = 0; i < dps.length; i++) {
-            mzValues[i] = dps[i].getMZ();
-            intensityValues[i] = dps[i].getIntensity();
+            mzValues[i] = mz;
+            intensityValues[i] = intensity;
           }
+          // data reading finished, apply data processing like sorting cropping mass detection if selected
+          var processedData = scanProcessorConfig.processor()
+              .processScan(buildingScan, new SimpleSpectralArrays(mzValues, intensityValues));
 
+          mzValues = processedData.mzs();
```

Comment:
why are we storing mzValues and intensityValues also separately from SimpleSpectralArrays?

Reply:
I wanted to keep the changes minimal and there are multiple places where we need the mz and intensities. Could also change all to the new class but then pass the arrays